context.go

```diff
@@ -153,6 +177,25 @@ func lookupFlag(name string, ctx *Context) Flag {
 	return nil
 }
 
+func (c *Context) resolveFlagDeep(name string) *flag.Flag {
+	var src *flag.Flag
+	for cur := c; cur != nil; cur = cur.parentContext {
+		if f := cur.flagSet.Lookup(name); f != nil {
+			if cur.flagOnCommandLine(name) {
+				// we've found most specific instance on command line, use it
+				src = f
+				break
+			}
+			if src == nil {
+				// flag was defined, but value is not present among flags of the current context
+				// remember the most specific instance of the flag not from command line as fallback
+				src = f
+			}
+		}
+	}
+	return src
+}
+
 func lookupFlagSet(name string, ctx *Context) *flag.FlagSet {
```

Comment:
Couldnt you incorporate the resolveFlagDeep logic into this function to retrieve the corresponding flagset ? That would have meant zero changes to existing flag_xxxx src files. Moreover the current code handles the case of lookupFlagSet returning nil which the proposed changes dont handle .